go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -186,21 +203,43 @@ func (h *secondaryLocalnetNetworkControllerEventHandler) IsObjectInTerminalState
 // for a secondary localnet network
 type SecondaryLocalnetNetworkController struct {
 	BaseSecondaryLayer2NetworkController
+	ovsClient          client.Client
+	ovsClientCancel    context.CancelFunc
+	controllerNodeName string
 }
 
 // NewSecondaryLocalnetNetworkController create a new OVN controller for the given secondary localnet NAD
 func NewSecondaryLocalnetNetworkController(
 	cnci *CommonNetworkControllerInfo,
 	netInfo util.NetInfo,
 	networkManager networkmanager.Interface,
-) *SecondaryLocalnetNetworkController {
+) (*SecondaryLocalnetNetworkController, error) {
+	return newSecondaryLocalnetNetworkController(cnci, netInfo, networkManager, nil)
+}
+
+func newSecondaryLocalnetNetworkController(
+	cnci *CommonNetworkControllerInfo,
+	netInfo util.NetInfo,
+	networkManager networkmanager.Interface,
+	ovsClient client.Client,
+) (*SecondaryLocalnetNetworkController, error) {
 
 	stopChan := make(chan struct{})
 
+	ctx, cancel := context.WithCancel(context.Background())
+	if ovsClient == nil {
+		var err error
+		ovsClient, err = libovsdb.NewOVSClient(ctx.Done())
```

Comment:
We should be using the OVS client that we already initialize in `ovnkube.go`. It brings a cache with it and we don't want dup caches.